nat-lab/tests/uniffi/libtelio_remote.py

```diff
@@ -0,0 +1,172 @@
+import base64
+import datetime
+import os
+import Pyro5.api  # type: ignore
+import Pyro5.server  # type: ignore
+import sys
+import telio_bindings as libtelio  # type: ignore
+import time
+from typing import List
+
+REMOTE_LOG = "remote.log"
+TCLI_LOG = "tcli.log"
+
+
+def serialize_error(f):
+    def wrap(*args, **kwargs):
+        with open(REMOTE_LOG, "a") as logfile:
+            fname = str(f).split(" ")[1]
+            args_str = ", ".join(map(str, args[1:]))
+            try:
+                res = f(*args, **kwargs)
+                logfile.write(
+                    f"{datetime.datetime.now()} {fname}({args_str}) => {res}\n"
+                )
+                return (res, None)
+            except libtelio.TelioError as e:
+                logfile.write(
+                    f"{datetime.datetime.now()} {fname}({args_str}) => {str(e)}\n"
+                )
+                return (None, str(e))
+
+    return wrap
+
+
+class TelioEventCbImpl(libtelio.TelioEventCb):
+    def __init__(self):
+        self._events = []
+
+    def event(self, payload):
+        self._events.append(payload)
+
+    def next_event(self):
+        if len(self._events) > 0:
+            return self._events.pop(0)
+        return None
+
+
+class TelioLoggerCbImpl(libtelio.TelioLoggerCb):
+    def log(self, log_level, payload):
```

Comment:
More of a question, but would it be possible to turn this into a callback that would work via pyro boundary?

Reply:
I haven't tried it so I cant say 100%, but I think it would work